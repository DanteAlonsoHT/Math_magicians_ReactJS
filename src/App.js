import Calculator from './components/Calculator';
import { useState } from 'react';

// eslint-disable-next-line react/prefer-stateless-function
const App = () => {
  const [objCalculator, setObt] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const updateState = (obj) => {
    setObt(obj);
  };

  render() {
    const { objCalculator } = this.state;
    return (
      <body><Calculator updateState={this.updateState} objCalculator={objCalculator} /></body>
    );
  }
}

export default App;
