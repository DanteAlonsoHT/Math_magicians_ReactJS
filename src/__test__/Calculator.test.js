import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from "../App"

beforeEach(() => {
  render(<App />);
});

describe('Test Calculator Result', () => {
  test('Result to be in the document', () => {
    const elementSeven = screen.getByTestId('math-result');
    expect(elementSeven).toBeInTheDocument();
  });
});