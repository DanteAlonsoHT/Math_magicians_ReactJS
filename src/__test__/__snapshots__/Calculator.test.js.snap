// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Calculator snapshot test 1`] = `
<DocumentFragment>
    <div className="section-content">
      <div className="title-calculator">
        <h4>Lets do some math!</h4>
      </div>
      <div className="calculator-content">
        <p className="quantity-field" data-testid="math-result">
          {showResult()}
        </p>
        <div className="row-buttons">
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, 'AC'); }}>AC</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '+/-'); }}>+/-</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '%'); }}>%</button>
          <button type="button" className="btn btn-orange" onClick={() => { calculatorButton(objCalculator, 'รท'); }}>รท</button>
        </div>
        <div className="row-buttons">
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '7'); }}>7</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '8'); }}>8</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '9'); }}>9</button>
          <button type="button" className="btn btn-orange" onClick={() => { calculatorButton(objCalculator, 'x'); }}>x</button>
        </div>
        <div className="row-buttons">
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '4'); }}>4</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '5'); }}>5</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '6'); }}>6</button>
          <button type="button" className="btn btn-orange" onClick={() => { calculatorButton(objCalculator, '-'); }}>-</button>
        </div>
        <div className="row-buttons">
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '1'); }}>1</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '2'); }}>2</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '3'); }}>3</button>
          <button type="button" className="btn btn-orange" onClick={() => { calculatorButton(objCalculator, '+'); }}>+</button>
        </div>
        <div className="row-buttons">
          <button type="button" className="big-btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '0'); }}>0</button>
          <button type="button" className="btn btn-gray-light" onClick={() => { calculatorButton(objCalculator, '.'); }}>.</button>
          <button type="button" className="btn btn-orange" onClick={() => { calculatorButton(objCalculator, '='); }}>=</button>
        </div>
      </div>
    </div>
</DocumentFragment>
`;
